src/syft/core/node/common/action/run_class_method_action.py

```diff
@@ -85,9 +86,25 @@ def execute_action(self, node: AbstractNode, verify_key: VerifyKey) -> None:
 
         result = method(resolved_self.data, *resolved_args, **resolved_kwargs)
 
-        # ensure that result has the right id
-        # TODO: overload all methods to incorporate this automatically
-        result.id = self.id_at_location
+        if isprimitive(value=result):
+            # Wrap in a PythonPrimitive
+            primitive = PythonPrimitive(value=result, id=self.id_at_location)
```

Comment:
MAYBE:
```
result = PythonPrimitive(value=result, id = self.id_at_location)
```
and remove
```result = StorableObject```
since we have one at line 110

Reply:
So PythonPrimitive IS an instance of StorableObject, so that code wont execute.

Reply:
So I think its normally done magically by this:
```
aggressive_set_attr(
    obj=th.nn.parameter.Parameter,
    name="serializable_wrapper_type",
    attr=PyTorchParameterWrapper,
)
```

But we arent injecting that into the primitive classes.

Reply:
Hmm...if we add the ```serializable_wrapper_type``` to the ```PythonPrimitive``` we still can't remove the result = ```StorableObject``` (from this ```if```).

(sorry if this is a stupid question)